### PR TITLE
Update README to Follow `terraform-aws-module-template`

### LINKED ISSUE #14: Update README.md
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* ***If you have submitted a pull request addressing the same issue, please leave a comment!***

***

### Description

In order to increase collaboration and contribution on the development of this module, it is a good idea to start with enhancing the README file. We do have a template repository which contains several files that will become a standard: https://github.com/traveloka/terraform-aws-modules-template

On that repository you will find a [README](https://github.com/traveloka/terraform-aws-modules-template/blob/master/README.md) file that can be used as an example. What points should be written on the README.

The example [README](https://github.com/traveloka/terraform-aws-modules-template/blob/master/README.md) was created by @Rukaan and reviewed by @rafikurnia, @sumodirjo, and also other @traveloka/site-infra members. See traveloka/terraform-aws-modules-template#9 for more information.

### References

* https://opensource.guide/starting-a-project/

### PR DESCRIPTION
**Pull request by rizkidoank**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

***

Fixes #14 

***

Release note for [CHANGELOG](https://github.com/traveloka/terraform-aws-vpc/blob/master/CHANGELOG.md):

```release-note
NOTES:

* Add pre-commit terraform hooks for `terraform fmt` and `terraform-docs`
* Update README to follow [terraform-aws-module-template](https://github.com/traveloka/terraform-aws-modules-template)
```

***

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+- [Dependencies](#Dependencies)
+- [Contributing](#Contributing)
+- [Contributor](#Contributor)
+- [License](#License)
+- [Acknowledgments](#Acknowledgments)
+
+## Prerequisites
+
+- [Terraform](https://releases.hashicorp.com/terraform/). This module currently tested on `0.11.14`
+- [awsudo](https://github.com/makethunder/awsudo) to assume role in AWS
```
Comment (sshvetsov) [inline README.md]: Is this really a prerequisite for the module, or is it more of our internal usage pattern?

File: README.md [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
+- [Visual Studio Code](https://code.visualstudio.com/download) is the best editor for the [Terraform Extension](https://marketplace.visualstudio.com/items?itemName=mauve.terraform). After install activate auto format by go to `File`→`Preferences`→`Settings`. Choose`Text Editor`→`Formatting` and check `Format on Save`
```
Comment (sshvetsov) [inline README.md]: I think opinions don't belong in module docs.

File: README.md [hunk 3]
```diff
@@ -1,11 +1,38 @@
...
+- `SuperAdmin` role, or role with VPC administrator access (read-write)
```
Comment (sshvetsov) [inline README.md]: Too Traveloka specific, should be removed.

File: README.md [hunk 4]
```diff
@@ -42,24 +81,101 @@ Currrently Terraform does not allow `count` inside `output` block, so now it is
...
-At the moment, this module is not supported on Terraform 0.12
 
-## Examples
+## Contributing
 
-* [Multi-Tier VPC](https://github.com/traveloka/terraform-aws-vpc/tree/master/examples/multi-tier)
+See [CONTRIBUTING.md](CONTRIBUTING.md)
 
-## Author
+## Contributor
```
Comment (sshvetsov) [inline README.md]: Can this part be auto-generated based on commits? Alternatively, should we remove this section and simply rely on GitHub's list of contributors?

File: README.md [hunk 5]
```diff
@@ -42,24 +81,101 @@ Currrently Terraform does not allow `count` inside `output` block, so now it is
...
 
 * [Rafi Kurnia Putra](https://github.com/rafikurnia)
 
 ## License
 
 Apache 2 Licensed. See LICENSE for full details.
+
+## Acknowledgement
```
Comment (sshvetsov) [inline README.md]: Missing trailing new line.

File: README.md (hunk 1 above)
Comment (rizkidoank) [reply README.md]: I have been discuss this with @rafikurnia , and confirmed its specifically for internal use.

File: README.md (hunk 2 above)
Comment (rizkidoank) [reply README.md]: Yes, I agree.
I will take a look at [terraform-aws-module-template](https://github.com/traveloka/terraform-aws-modules-template) to update the template

File: README.md (hunk 4 above)
Comment (rizkidoank) [reply README.md]: IMO, we could remove this and also any section that we can actually get the information from repository information. i.e: contributor, versioning.

**Comment by rizkidoank** [comment]
@rafikurnia lets revisit https://github.com/traveloka/terraform-aws-modules-template and update the template accordingly.

File: README.md (hunk 1 above)
Comment (sshvetsov) [reply README.md]: This is a public module, though... If this module is for internal use only, should it be in a public repo?

File: README.md (hunk 4 above)
Comment (sshvetsov) [reply README.md]: Perhaps we should then. I believe contributors/maintainers will change over time, so this section would be misleading.

File: README.md (hunk 1 above)
Comment (rizkidoank) [reply README.md]: what I have confirmed was about the prerequisites (awsudo, vscode and terraform extension) from readme template, not the module. The module is applicable for internal as well as public.

File: README.md [hunk 6]
```diff
@@ -1,11 +1,34 @@
 # terraform-aws-vpc
 
+[![Terraform Version](https://img.shields.io/badge/Terraform%20Version-0.11.14-blue.svg)](https://releases.hashicorp.com/terraform/)
```
Comment (rafikurnia) [inline README.md]: How about specifying range instead?
Example:
[Terraform Version](https://releases.hashicorp.com/terraform/)

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+- repo: git://github.com/antonbabenko/pre-commit-terraform
+  rev: v1.22.0
```
[the remark this answers is not in the document]
Comment (rizkidoank) [reply .pre-commit-config.yaml]: Previously use latest version, but somehow doesn't work (didn't read the configs). It works when I use `v1.22.0`.

**Pull request merged**